lua/lspconfig/serve_d.lua

```diff
@@ -0,0 +1,22 @@
+local configs = require 'lspconfig/configs'
+local util = require 'lspconfig/util'
+
+configs.serve_d = {
+  default_config = {
+    cmd = { 'serve-d' },
+    filetypes = { 'd' },
+    root_dir = function(fname)
+      return util.root_pattern('dub.json', 'dub.sdl', '.git')(fname)
```

Comment:
if you're just going to pass fname to a single helper, this simplifies to 

root_dir = util.root_pattern('dub.json', 'dub.sdl', '.git')